src/puppetlabs/kitchensink/core.clj

```diff
@@ -944,6 +944,11 @@ to be a zipper."
   []
   (str (java.util.UUID/randomUUID)))
 
+(defn uuid?
+  "Verifies whether a string is a valid UUID"
+  [uuid]
+  (re-matches #"^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$" uuid))
```

Comment:
I'd guess that `fromString` is faster than a regex, but the exception handling might slow it down.  Would be interesting to test, though obviously not important.

Anywho, this LGTM now.